diff --git a/Makefile b/Makefile
index c80b805..a45fb61 100644
--- a/Makefile
+++ b/Makefile
@@ -102,9 +102,6 @@ endif
 
 $(KRUNFW_BINARY_$(OS)): $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
 	$(CC) -fPIC -DABI_VERSION=$(ABI_VERSION) -shared $(SONAME_$(OS)) -o $@ $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
-ifeq ($(OS),Linux)
-	$(STRIP) $(KRUNFW_BINARY_$(OS))
-endif
 
 install:
 	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR_$(OS))/
