diff --git a/Makefile b/Makefile
index df3c206..6526e7d 100644
--- a/Makefile
+++ b/Makefile
@@ -90,9 +90,6 @@ endif
 
 $(KRUNFW_BINARY_$(OS)): $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
 	$(CC) -fPIC -DABI_VERSION=$(ABI_VERSION) -shared $(SONAME_$(OS)) -o $@ $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
-ifeq ($(OS),Linux)
-	strip $(KRUNFW_BINARY_$(OS))
-endif
 
 install:
 	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR_$(OS))/
