diff --git a/Makefile b/Makefile
index b9d2bef..9039f3a 100644
--- a/Makefile
+++ b/Makefile
@@ -57,7 +57,6 @@ all: $(KRUNFW_BINARY_$(OS))
 
 $(KERNEL_SOURCES):
 	for patch in $(KERNEL_PATCHES); do patch -p1 -d $(KERNEL_SOURCES) < "$$patch"; done
-	cp config-libkrunfw$(VARIANT)_$(ARCH) $(KERNEL_SOURCES)/.config
 	cd $(KERNEL_SOURCES) ; $(MAKE) ARCH=$(KARCH) olddefconfig
 
 $(KERNEL_BINARY_$(ARCH)): $(KERNEL_SOURCES)
