diff --git a/setup.py b/setup.py
index 1b371ba..c18acca 100644
--- a/setup.py
+++ b/setup.py
@@ -9,5 +9,5 @@ setup(name='asahi_firmware',
       author_email='marcan@marcan.st',
       url='https://github.com/AsahiLinux/asahi-installer/',
       packages=['asahi_firmware'],
-      entry_points={"console_scripts": ["asahi-fwextract = asahi_firmware.update:main"]}
+      entry_points={"console_scripts": ["asahi-fw-module = asahi_firmware.update:main"]}
      )
