commit 763000de8d4f8aee1eb0bc0cf36f3f45fba07a1f
Author: Sasha Finkelstein <fnkl.kernel@gmail.com>
Date:   Wed Aug 21 13:31:16 2024 +0200

    unvendor drm-headers

diff --git a/Source/Tools/LinuxEmulation/CMakeLists.txt b/Source/Tools/LinuxEmulation/CMakeLists.txt
index 3076a31..6deb0e6 100644
--- a/Source/Tools/LinuxEmulation/CMakeLists.txt
+++ b/Source/Tools/LinuxEmulation/CMakeLists.txt
@@ -72,7 +72,6 @@ target_include_directories(LinuxEmulation
 PRIVATE
   ${CMAKE_BINARY_DIR}/generated
   ${CMAKE_CURRENT_SOURCE_DIR}/
-  ${PROJECT_SOURCE_DIR}/External/drm-headers/include/
 )
 
 target_include_directories(LinuxEmulation
diff --git a/Source/Tools/LinuxEmulation/LinuxSyscalls/x32/Ioctl/drm.h b/Source/Tools/LinuxEmulation/LinuxSyscalls/x32/Ioctl/drm.h
index efecf2c..cded3a0 100644
--- a/Source/Tools/LinuxEmulation/LinuxSyscalls/x32/Ioctl/drm.h
+++ b/Source/Tools/LinuxEmulation/LinuxSyscalls/x32/Ioctl/drm.h
@@ -10,21 +10,21 @@ extern "C" {
 // drm headers use a `__user` define that has an address_space attribute. This allows their tooling to see unsafe user-space accesses.
 // Define this to nothing so we don't need to modify those headers.
 #define __user
-#include "fex-drm/drm.h"
-#include "fex-drm/drm_mode.h"
-#include "fex-drm/i915_drm.h"
-#include "fex-drm/amdgpu_drm.h"
-#include "fex-drm/lima_drm.h"
-#include "fex-drm/panfrost_drm.h"
-#include "fex-drm/msm_drm.h"
-#include "fex-drm/nouveau_drm.h"
-#include "fex-drm/radeon_drm.h"
-#include "fex-drm/vc4_drm.h"
-#include "fex-drm/v3d_drm.h"
-#include "fex-drm/panthor_drm.h"
-#include "fex-drm/pvr_drm.h"
-#include "fex-drm/virtgpu_drm.h"
-#include "fex-drm/xe_drm.h"
+#include <drm/drm.h>
+#include <drm/drm_mode.h>
+#include <drm/i915_drm.h>
+#include <drm/amdgpu_drm.h>
+#include <drm/lima_drm.h>
+#include <drm/panfrost_drm.h>
+#include <drm/msm_drm.h>
+#include <drm/nouveau_drm.h>
+#include <drm/radeon_drm.h>
+#include <drm/vc4_drm.h>
+#include <drm/v3d_drm.h>
+#include <drm/panthor_drm.h>
+#include <drm/pvr_drm.h>
+#include <drm/virtgpu_drm.h>
+#include <drm/xe_drm.h>
 }
 #include <sys/ioctl.h>
 
diff --git a/Source/Tools/LinuxEmulation/LinuxSyscalls/x64/Ioctl/drm.h b/Source/Tools/LinuxEmulation/LinuxSyscalls/x64/Ioctl/drm.h
index ae8b0fa..ddb565f 100644
--- a/Source/Tools/LinuxEmulation/LinuxSyscalls/x64/Ioctl/drm.h
+++ b/Source/Tools/LinuxEmulation/LinuxSyscalls/x64/Ioctl/drm.h
@@ -7,17 +7,17 @@
 
 #include <cstdint>
 extern "C" {
-#include "fex-drm/drm.h"
-#include "fex-drm/drm_mode.h"
-#include "fex-drm/i915_drm.h"
-#include "fex-drm/amdgpu_drm.h"
-#include "fex-drm/lima_drm.h"
-#include "fex-drm/panfrost_drm.h"
-#include "fex-drm/msm_drm.h"
-#include "fex-drm/nouveau_drm.h"
-#include "fex-drm/vc4_drm.h"
-#include "fex-drm/v3d_drm.h"
-#include "fex-drm/virtgpu_drm.h"
+#include <drm/drm.h>
+#include <drm/drm_mode.h>
+#include <drm/i915_drm.h>
+#include <drm/amdgpu_drm.h>
+#include <drm/lima_drm.h>
+#include <drm/panfrost_drm.h>
+#include <drm/msm_drm.h>
+#include <drm/nouveau_drm.h>
+#include <drm/vc4_drm.h>
+#include <drm/v3d_drm.h>
+#include <drm/virtgpu_drm.h>
 }
 #include <sys/ioctl.h>
 
