diff --git a/External/fmt/CMakeLists.txt b/External/fmt/CMakeLists.txt
index e3b35110..58ada4da 100644
--- a/External/fmt/CMakeLists.txt
+++ b/External/fmt/CMakeLists.txt
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.8...3.28)
 
+set(BUILD_SHARED_LIBS OFF)
+
 # Fallback for using newer policies on CMake <3.12.
 if (${CMAKE_VERSION} VERSION_LESS 3.12)
   cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
